### PR TITLE
Atan2 operator for LazyTensors

### PR DESCRIPTION
**Pull request by haguettaz**
Adds atan2 operation, similar to PyTorch's implementation.

At the moment, I don't know how to deal with the gradient of atan2 since it depends on x and y. Any idea?

Test:
```
import pykeops
import torch
from pykeops.torch import LazyTensor

device = 'cpu' 
eps = 1e-3
x = torch.FloatTensor(1000, 1).uniform_(-1, 1)
y = torch.FloatTensor(1000, 1).uniform_(-1, 1)
x_ = x.clone()
y_ = y.clone()
x = x.to(device)
y = y.to(device)
x_ = x_.to(device)
y_ = y.to(device)
x_.requires_grad = True
y_.requires_grad = True

x_i = LazyTensor(x[:, None])
y_i = LazyTensor(y[:, None])
s1 = y_i.atan2(x_i).sum(0)
s2 = torch.atan2(y_, x_).sum()
print(torch.abs(s1 - s2))
assert torch.abs(s1 - s2) < 1e-3
```

### CODE REVIEW

**Comment by joanglaunes** [comment]
Hello @haguettaz ,
Thank you for this contribution ; I have now merged it into master after having edited it to benefit from @fwilliams's version in #147 which gives also the gradient.
